novice-goals.Rmd

```diff
@@ -0,0 +1,207 @@
+# Outline {#outline}
+
+This outline describes the questions that the novice courses on R and Python will answer.
+The advanced course can then assume that learners have hands-on experience with these topics
+but nothing more.
+
+## Data manipulation
+
+-   How can I read tabular data into a program?
+    -   CSV
+    -   reading files
+-   How can I select subsets of my data?
+    -   select
+    -   filter
+    -   Boolean conditions
+-   How can I calculate new values?
+    -   mutate
+    -   ifelse
+-   How can I operate on subsets of my data?
+    -   group
+    -   summarize
+-   How can I work with two or more datasets?
+    -   join
+-   What *isn't* included?
+    -   anything other than reasonably tidy tabular data
+    -   map
+    -   loops and conditionals
+
+## Programming
+
+-   What are the different ways I can interact with software?
+    -   console
+    -   scripts
+    -   libraries/packages?
+-   How can I find and view help?
+    -   In the IDE
+    -   Stack Overflow
+-   How can I ask for help?
+    -   creating a reproducible example (reprex)
+-   How do I install software?
+    -   package manager
+-   How can I inspect data while I'm working on it?
+    -   table viewers
+-   How can I tell what's gone wrong in my programs?
+    -   reading error messages
+-   What *isn't* included?
+    -   code browsers, multiple cursors, and other fancy IDE tricks
+    -   virtual environments
+    -   debuggers
+
+## Plotting
+
+-   What are the core elements of every plot?
+    -   data
+    -   geometry
+    -   mapping
+-   What kinds of plot should I use?
+    -   Abela's Rules
+-   How can I create different kinds of plots?
+    -   scatter plot
+    -   line plot
+    -   histogram
+-   How can I plot multiple datasets at once?
+    -   grouping
+    -   faceting
+-   How can I make misleading plots?
+    -   examples of what not to do
+-   What *isn't* included?
+    -   interactive plots
+    -   maps
+    -   3D visualization
+
+## Data analysis
+
+-   How can I represent and manage missing values?
+    -   NA
+-   How can I get a feel for my data?
+    -   summary statistics
+-   How can I create a simple model of my data?
+    -   formulas
+    -   linear regression
+-   How can I put people at risk?
+    -   algorithmic bias
+    -   de-anonymization
+-   What *isn't* included?
+    -   statistical tests
+    -   anything with "machine learning" in its name
+
+## Verification
+
+-   What is the difference between validation and verification?
+    -   did we build the right thing vs. did we build it correctly
+-   How can I make my programs tell me that something has gone wrong?
+    -   assertions
+-   How can I write and manage tests for my software?
+    -   unit tests
+    -   test runners
+-   What tests should I write for my software?
+    -   edge cases
+    -   fails when it should
+    -   bugs become tests
+-   What *isn't* included?
+    -   legal liabilities
```

Comment:
add - Has this program provided the answers we are seeking? What are our conclusions?

Reply:
Do you mean "add to the list of things we're not including in the novice course" or "add as a major point that we should cover"? I'm assuming the former...

Reply:
Yes, I meant to add it to the former.

Reply:
:+1: